### PR TITLE
[None][fix] share Kimi auxiliary streams

### PR DESCRIPTION
**Pull request by jiaganc**
## Dev Engineer Review

- Added a model-owned auxiliary-stream registry for Kimi K3.
- Reused one physical stream for sequential attention and shared-expert execution.
- Kept separate streams for MoE chunking, load balancing, and output initialization.
- Propagated `aux_stream_dict` through KDA, MLA, and MoE constructors.
- Pre-registered the `MoeBalancer` stream for future EPLB support.
- No configuration or test-list files changed.
- Reported validation includes pre-commit checks, a CUDA architecture 100 B200 build, and targeted Kimi, MLA, and MoE tests.
- Verdict: sufficient.

## QA Engineer Review

- Added `test_kimi_linear_model_builds_shared_aux_stream_registry`.
- Updated `test_kimi_k3_shared_expert_parallel_construction`.
- Updated Kimi K3 attention stream assertions in `test_kimi_k3_mlp.py`.
- `tests/integration/test_lists/test-db/l0_h100.yml` and `l0_b200.yml` cover `test_kimi_k3_mlp.py`.
- No test-list entry covers `test_kimi_linear_modeling.py`.
- Verdict: needs follow-up.

## Description

Kimi K3 previously passed one model-owned side stream to KDA and the shared
expert, while MLA and routed MoE backends could allocate auxiliary streams
independently for each layer.

This change introduces a model-owned auxiliary-stream registry shared by every
decoder layer. Attention and shared-expert overlap reuse one physical stream
because those phases execute sequentially, while MoE chunking, load balancing,
and output memset retain separate streams. The registry is passed through KDA,
MLA, and `create_moe`, and pre-registers the `MoeBalancer` stream needed by
future EPLB support.

## Test Coverage

- `pre-commit run --from-ref upstream/main --to-ref HEAD`: passed on the current commit.
- Incremental CUDA architecture 100 build on B200: passed before rebasing onto current `main`.
- Targeted Kimi model, MLA policy, and MoE unit tests on B200: 27 passed before rebasing.
- Model stream-registry unit test on B200: 1 passed before rebasing.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Reviewed the checklist above; no public API, dependency, ownership, documentation, or architecture-diagram update is required.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by mikeiovine — APPROVED** [review]
Stamp on behalf of runtime devs, delegating proper review to @NVIDIA/trt-llm-models-devs; please ping me if you think this is not accurate

**Review by sylvesterkaczmarek — COMMENTED** [review]
Centralising stream ownership at the model level is the right lifetime model: every decoder layer now shares one registry, while roles that can overlap physically keep distinct streams. Reusing the Attention stream for `MoeShared` is valid only because those phases are strictly sequential; the identity assertions are therefore important because accidental future overlap would become a correctness issue, not just a performance regression. I would keep that sequencing assumption close to the scheduling code as well as in the allocation comment.

**Review by leslie-fang25 — APPROVED** [review]

File: tests/unittest/_torch/modules/test_kimi_k3_mla_backend.py [hunk 1]
```diff
@@ -130,6 +130,7 @@ def test_kimi_k3_mla_construction_fails_fast_for_fp8_kv_with_attention_dp(
             layer_idx=0,
             max_position_embeddings=256,
             model_config=model_config,
+            aux_stream_dict={},
```
Comment (Wanli-Jiang) [inline tests/unittest/_torch/modules/test_kimi_k3_mla_backend.py:133]: passed an empty value, which can not trigger the real meaningful tests. can we add the real multi-stream test for kimi-k3-mla?

**Review by Wanli-Jiang — APPROVED** [review]

File: tests/unittest/_torch/modules/test_kimi_k3_mla_backend.py (hunk 1 above)
Comment (jiaganc) [reply tests/unittest/_torch/modules/test_kimi_k3_mla_backend.py]: Will not change this test because this is a neagtive config test, and this test will be removed in https://github.com/NVIDIA/TensorRT-LLM/pull/18653. As for adding real test, seems like there's no real unit test at all for K3's MLA module, so will add in the following PR. Create a JIRA task to track: https://jirasw.nvidia.com/browse/TRTLLM-16202.

File: tests/unittest/_torch/modeling/test_kimi_linear_modeling.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import torch
+from torch import nn
+
+from tensorrt_llm._torch.configs.kimi_linear import KimiLinearConfig
+from tensorrt_llm._torch.model_config import ModelConfig
+from tensorrt_llm._torch.models import modeling_kimi_linear
+from tensorrt_llm._torch.utils import AuxStreamType
+
+
+def test_kimi_linear_model_builds_shared_aux_stream_registry(
```
Comment (aswinvisva) [inline tests/unittest/_torch/modeling/test_kimi_linear_modeling.py:16]: I don't think this new test will actually run in CI, should we add to the appropriate CI list?

**Review by aswinvisva — APPROVED** [review]

File: tests/unittest/_torch/modeling/test_kimi_linear_modeling.py (hunk 2 above)
Comment (jiaganc) [reply tests/unittest/_torch/modeling/test_kimi_linear_modeling.py:16]: Good catch. The CPU CI list already includes `unittest/_torch/modeling`, but its collector ignores files without a `cpu_only` marker. Added `pytestmark = pytest.mark.cpu_only` in 8d106bcf53 so the existing CPU stage will collect this test; no separate test-list entry is needed.
